### PR TITLE
Laboratory work #1, Maria Bukina - 22FPL2

### PR DESCRIPTION
**Pull request by MaryBukina**
Looking forward to get 6!

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -11,6 +11,16 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+
+    tokens = []
+    for i in text:
+        if i.isalpha():
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:19]: list comprehension

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -20,6 +30,18 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :return: a dictionary with frequencies
     """
 
+    if not isinstance(tokens, list):
+        return None
+
+    dictionary = {}
+    tokenQuantity = len(tokens)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:37]: snake_case

File: lab_1_classify_by_unigrams/start.py [hunk 3]
```diff
@@ -9,12 +10,16 @@ def main() -> None:
     """
     with open("assets/texts/en.txt", "r", encoding="utf-8") as file_to_read_en:
         en_text = file_to_read_en.read()
+        tokens = tokenize(en_text)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/start.py:13]: read is only one operation in this case that must be in with

Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/start.py:13]: read is only one operation in this case that must be in with

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Comment by MaryBukina** [comment]
@artyomtugaryov, changes have been made :)

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -11,6 +11,13 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+
+    tokens = [x.lower() for x in text if x.isalpha()]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:17]: return as is, do not create the variable

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -20,6 +27,19 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        return None
+
+    dictionary = {}
+    token_quantity = len(tokens)
+    for i in tokens:
+        if not isinstance(i, str):
+            return None
+        if i not in dictionary:
+            dictionary[i] = tokens.count(i) / token_quantity
+            pass
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:40]: For what is this pass?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (MaryBukina) [reply lab_1_classify_by_unigrams/main.py:40]: this pass is only for pointing it while debug as it gives the result dictionary

**Pull request closed without merging**